api/lib/application/authentication/authentication-controller.js

```
@@ -6,21 +6,22 @@ module.exports = {
   /**
    * @see https://tools.ietf.org/html/rfc6749#section-4.3
    */
-  authenticateUser(request, h) {
+  async authenticateUser(request, h) {
     const { username, password, scope } = request.payload;
 
-    return usecases.authenticateUser({ username, password, scope })
-      .then((accessToken) => {
-        return h.response({
-          token_type: 'bearer',
-          access_token: accessToken,
-          user_id: tokenService.extractUserId(accessToken),
-        })
-          .code(200)
-          .header('Content-Type', 'application/json;charset=UTF-8')
-          .header('Cache-Control', 'no-store')
-          .header('Pragma', 'no-cache');
-      });
-  },
+    const accessToken = await usecases.authenticateUser({ username, password, scope });
+
+    if (accessToken) {
```

Il ne manque pas une valeur de retour dans la fonction si on n'a pas d'accessToken ?

---

Le use-case peut lever des erreurs, mais elles sont gérées correctement par le domaine:
- UserShouldChangePasswordError
- ForbiddenAccess
- MissingOrInvalidCredentialsError

J'ai regardé d'autres controller (ex: password, membership), et aucun cas d'erreur n'est implémenté. Je pense que c'est intentionnel: aucune erreur ne doit être gérée par le controller (et si un problème arrive, ce sera du fail fast => HAPI va le gérer)

---

Oui mais là il ne s'agit pas des cas où une exception est levée par le _usecase_, mais plutôt des cas (a priori inexistants) où le _usecase_ renverrait avec succès une valeur vide (_falsy_). 

Si `accessToken` est vide `authenticateUser` retournera `undefined` et `hapi` émettra une erreur obscure parce qu'une méthode du contrôleur n'a pas le droit de renvoyer `undefined`.

Dit autrement, ce code n'est pas équivalent à celui d'avant qui utilisait `.then()`. Pour avoir le code équivalent il faudrait simplement enlever ce `if` qui n'existait pas à l'origine.

---

Ok, on corrige !

---

Done